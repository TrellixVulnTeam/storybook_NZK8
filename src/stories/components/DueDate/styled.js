// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const DueDateWrapper = styled.div`
  color: ${props => (props.pastDue ? colors.strawberry : colors.slateGrey)};
  font-weight: ${props => (props.pastDue ? 600 : 400)};

  svg {
    margin: 0 4px;
  }
`;

export const EmptyTableValue = styled.div`
  width: 58px;
  text-align: center;
`;

export const DueDateIcons = styled.span`
  @media (max-width: 875px) {
    display: none;
  }
`;
