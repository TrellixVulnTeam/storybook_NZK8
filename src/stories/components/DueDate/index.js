// @flow

import React from 'react';
import moment from 'moment';
import Tooltip from 'components/Tooltip';
import { DueDateWrapper, DueDateIcons, EmptyTableValue } from './styled';

import RepeatIcon from './img/repeat.svg';
import AlertIcon from './img/alert.svg';

type Props = {
  dueDate: string,
  schedule: Object,
  status: string,
  id?: string,
};

export default (props: Props) => {
  const { dueDate, schedule, status } = props;
  const isDue = moment(dueDate).isBefore(moment(), 'now');
  const isRepeating = schedule !== null;
  const isComplete = status === 'COMPLETE';
  const formattedDate = moment(dueDate).format(localStorage.getItem('dateFormat') || 'MM/DD/YY');

  /**
   * @return {Array} - Array of status icons
   */
  const renderStatusIcon = () => {
    const icons = [];
    let id = props.id ? `${props.id}-alert` : '';
    if (isDue && !isComplete) {
      icons.push(
        <Tooltip value="Past Due" id={id} place="top" effect="solid">
          <AlertIcon key="alert" />
        </Tooltip>,
      );
    }
    if (isRepeating) {
      id = props.id ? `${props.id}-repeat` : '';
      icons.push(
        <Tooltip value="Repeating" id={id} place="top" effect="solid">
          <RepeatIcon key="repeating" />
        </Tooltip>,
      );
    }
    return icons;
  };

  return (
    <DueDateWrapper pastDue={isDue && !isComplete}>
      {dueDate ? (
        <div>
          {formattedDate}{' '}
          <DueDateIcons>{(isDue || isRepeating) && renderStatusIcon()}</DueDateIcons>
        </div>
      ) : (
        <EmptyTableValue>-</EmptyTableValue>
      )}
    </DueDateWrapper>
  );
};
