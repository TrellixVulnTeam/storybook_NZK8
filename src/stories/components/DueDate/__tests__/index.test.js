import React from 'react';
import { mount } from 'enzyme';
import DueDate from 'components/DueDate/index';

describe('<DueDate />', () => {
  const mandatoryProps = {
    dueDate: '0',
    schedule: {
      some: 'thing',
    },
    status: 'COMPLETE',
  };

  it('should render correctly with mandatory props', () => {
    const sut = mount(<DueDate {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with id', () => {
    const props = {
      ...mandatoryProps,
      id: 1,
    };
    const sut = mount(<DueDate {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly without schedule', () => {
    const props = {
      ...mandatoryProps,
      schedule: null,
    };
    const sut = mount(<DueDate {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly without due date', () => {
    const props = {
      ...mandatoryProps,
      dueDate: false,
    };
    const sut = mount(<DueDate {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if not completed', () => {
    const props = {
      ...mandatoryProps,
      status: '',
    };
    const sut = mount(<DueDate {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
