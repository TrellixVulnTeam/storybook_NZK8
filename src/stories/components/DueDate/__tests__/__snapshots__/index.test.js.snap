// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DueDate /> should render correctly if not completed 1`] = `
<_default
  dueDate="0"
  schedule={
    Object {
      "some": "thing",
    }
  }
  status=""
>
  <styled__DueDateWrapper
    pastDue={true}
  >
    <div
      className="styled__DueDateWrapper-sc-1l6r9up-0 lhokGd"
    >
      <div>
        01/01/00
         
        <styled__DueDateIcons>
          <span
            className="styled__DueDateIcons-sc-1l6r9up-2 fcNomY"
          >
            <_default
              effect="solid"
              id=""
              place="top"
              value="Past Due"
            >
              <styled__Wrapper
                data-for=""
                data-tip="Past Due"
                effect="solid"
                id=""
                place="top"
                value="Past Due"
              >
                <div
                  className="styled__Wrapper-funpqb-0 cznzQQ"
                  data-for=""
                  data-tip="Past Due"
                  id=""
                  value="Past Due"
                >
                  <test-file-stub
                    key="alert"
                  />
                  <styled__TooltipWrapper
                    display="true"
                  >
                    <div
                      className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                      display="true"
                    >
                      <ReactTooltip
                        clickable={false}
                        delayShow={300}
                        effect="solid"
                        getContent={[Function]}
                        html={true}
                        id=""
                        insecure={true}
                        multiline={true}
                        place="top"
                        resizeHide={true}
                        type="dark"
                        wrapper="div"
                      >
                        <div
                          className="__react_component_tooltip place-top type-dark"
                          data-id="tooltip"
                          id=""
                        >
                          null
                        </div>
                      </ReactTooltip>
                    </div>
                  </styled__TooltipWrapper>
                </div>
              </styled__Wrapper>
            </_default>
            <_default
              effect="solid"
              id=""
              place="top"
              value="Repeating"
            >
              <styled__Wrapper
                data-for=""
                data-tip="Repeating"
                effect="solid"
                id=""
                place="top"
                value="Repeating"
              >
                <div
                  className="styled__Wrapper-funpqb-0 cznzQQ"
                  data-for=""
                  data-tip="Repeating"
                  id=""
                  value="Repeating"
                >
                  <test-file-stub
                    key="repeating"
                  />
                  <styled__TooltipWrapper
                    display="true"
                  >
                    <div
                      className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                      display="true"
                    >
                      <ReactTooltip
                        clickable={false}
                        delayShow={300}
                        effect="solid"
                        getContent={[Function]}
                        html={true}
                        id=""
                        insecure={true}
                        multiline={true}
                        place="top"
                        resizeHide={true}
                        type="dark"
                        wrapper="div"
                      >
                        <div
                          className="__react_component_tooltip place-top type-dark"
                          data-id="tooltip"
                          id=""
                        >
                          null
                        </div>
                      </ReactTooltip>
                    </div>
                  </styled__TooltipWrapper>
                </div>
              </styled__Wrapper>
            </_default>
          </span>
        </styled__DueDateIcons>
      </div>
    </div>
  </styled__DueDateWrapper>
</_default>
`;

exports[`<DueDate /> should render correctly with id 1`] = `
<_default
  dueDate="0"
  id={1}
  schedule={
    Object {
      "some": "thing",
    }
  }
  status="COMPLETE"
>
  <styled__DueDateWrapper
    pastDue={false}
  >
    <div
      className="styled__DueDateWrapper-sc-1l6r9up-0 eMWSAz"
    >
      <div>
        01/01/00
         
        <styled__DueDateIcons>
          <span
            className="styled__DueDateIcons-sc-1l6r9up-2 fcNomY"
          >
            <_default
              effect="solid"
              id="1-repeat"
              place="top"
              value="Repeating"
            >
              <styled__Wrapper
                data-for="1-repeat"
                data-tip="Repeating"
                effect="solid"
                id="1-repeat"
                place="top"
                value="Repeating"
              >
                <div
                  className="styled__Wrapper-funpqb-0 cznzQQ"
                  data-for="1-repeat"
                  data-tip="Repeating"
                  id="1-repeat"
                  value="Repeating"
                >
                  <test-file-stub
                    key="repeating"
                  />
                  <styled__TooltipWrapper
                    display="true"
                  >
                    <div
                      className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                      display="true"
                    >
                      <ReactTooltip
                        clickable={false}
                        delayShow={300}
                        effect="solid"
                        getContent={[Function]}
                        html={true}
                        id="1-repeat"
                        insecure={true}
                        multiline={true}
                        place="top"
                        resizeHide={true}
                        type="dark"
                        wrapper="div"
                      >
                        <div
                          className="__react_component_tooltip place-top type-dark"
                          data-id="tooltip"
                          id="1-repeat"
                        >
                          null
                        </div>
                      </ReactTooltip>
                    </div>
                  </styled__TooltipWrapper>
                </div>
              </styled__Wrapper>
            </_default>
          </span>
        </styled__DueDateIcons>
      </div>
    </div>
  </styled__DueDateWrapper>
</_default>
`;

exports[`<DueDate /> should render correctly with mandatory props 1`] = `
<_default
  dueDate="0"
  schedule={
    Object {
      "some": "thing",
    }
  }
  status="COMPLETE"
>
  <styled__DueDateWrapper
    pastDue={false}
  >
    <div
      className="styled__DueDateWrapper-sc-1l6r9up-0 eMWSAz"
    >
      <div>
        01/01/00
         
        <styled__DueDateIcons>
          <span
            className="styled__DueDateIcons-sc-1l6r9up-2 fcNomY"
          >
            <_default
              effect="solid"
              id=""
              place="top"
              value="Repeating"
            >
              <styled__Wrapper
                data-for=""
                data-tip="Repeating"
                effect="solid"
                id=""
                place="top"
                value="Repeating"
              >
                <div
                  className="styled__Wrapper-funpqb-0 cznzQQ"
                  data-for=""
                  data-tip="Repeating"
                  id=""
                  value="Repeating"
                >
                  <test-file-stub
                    key="repeating"
                  />
                  <styled__TooltipWrapper
                    display="true"
                  >
                    <div
                      className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                      display="true"
                    >
                      <ReactTooltip
                        clickable={false}
                        delayShow={300}
                        effect="solid"
                        getContent={[Function]}
                        html={true}
                        id=""
                        insecure={true}
                        multiline={true}
                        place="top"
                        resizeHide={true}
                        type="dark"
                        wrapper="div"
                      >
                        <div
                          className="__react_component_tooltip place-top type-dark"
                          data-id="tooltip"
                          id=""
                        >
                          null
                        </div>
                      </ReactTooltip>
                    </div>
                  </styled__TooltipWrapper>
                </div>
              </styled__Wrapper>
            </_default>
          </span>
        </styled__DueDateIcons>
      </div>
    </div>
  </styled__DueDateWrapper>
</_default>
`;

exports[`<DueDate /> should render correctly without due date 1`] = `
<_default
  dueDate={false}
  schedule={
    Object {
      "some": "thing",
    }
  }
  status="COMPLETE"
>
  <styled__DueDateWrapper
    pastDue={false}
  >
    <div
      className="styled__DueDateWrapper-sc-1l6r9up-0 eMWSAz"
    >
      <styled__EmptyTableValue>
        <div
          className="styled__EmptyTableValue-sc-1l6r9up-1 fTyxVG"
        >
          -
        </div>
      </styled__EmptyTableValue>
    </div>
  </styled__DueDateWrapper>
</_default>
`;

exports[`<DueDate /> should render correctly without schedule 1`] = `
<_default
  dueDate="0"
  schedule={null}
  status="COMPLETE"
>
  <styled__DueDateWrapper
    pastDue={false}
  >
    <div
      className="styled__DueDateWrapper-sc-1l6r9up-0 eMWSAz"
    >
      <div>
        01/01/00
         
        <styled__DueDateIcons>
          <span
            className="styled__DueDateIcons-sc-1l6r9up-2 fcNomY"
          />
        </styled__DueDateIcons>
      </div>
    </div>
  </styled__DueDateWrapper>
</_default>
`;
